nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -894,6 +894,8 @@ ngx_http_modsecurity_handler(ngx_http_request_t *r)
         ngx_http_set_ctx(r, ctx, ngx_http_modsecurity);
     }
 
+    modsecReopenLogfileIfNeeded(ctx->req);
```

Comment:
won't this give a compile error if not WAF_JSON_LOGGING_ENABLE?

Reply:
On a broader note, since we have separate branches for ModSecurity for IIS and Nginx, perhaps the whole `WAF_JSON_LOGGING_ENABLED` macro can be removed and made a default behavior?
Of course, not as a part of this fix and your point above is valid - if that needs to be wrapped with ifdef-s, let's do that.